arraycontext/impl/pytato/__init__.py

```diff
@@ -387,7 +387,19 @@ def get_target(self):
                     self.using_svm and dev.type & cl.device_type.GPU
                     and cl_char.has_coarse_grain_buffer_svm(dev))):
 
-            limit = dev.max_parameter_size
+            if dev.max_parameter_size == 4352:
+                # Nvidia devices and PTXAS advertise a limit of 4352 bytes,
+                # which is incorrect. The CUDA documentation at
+                # https://docs.nvidia.com/cuda/cuda-c-programming-guide/#function-parameters
+                # advertises a limit of 4KB, which is also incorrect.
+                # As far as I can tell, the actual limit is around 4080
+                # bytes, at least on a K40. Reducing the limit further
+                # in order ot be on the safe side.
+
+                limit = 4096-200
```

Comment:
Emit a warning that this is happening.

Reply:
Added in [1b94232](https://github.com/inducer/arraycontext/pull/198/commits/1b94232331f20911510a4b17a72e0f5579073c9a) (we could also make this a `logger.info` message)